feat(skills): add data-governance controls reference to privacy-standards (#2416)
microsoft/hve-core: .github/skills/project-planning/privacy-standards/references/data-governance-controls.md

---
title: Data governance controls for release readiness

> description: Operational data-governance controls — classification tiers, role-aware redaction, tiered retention, and tamper-evident audit — for production-readiness privacy planning

Reviewer: The description uses em dashes; the writing-style guide prefers parentheses or colons here.

```suggestion
description: Operational data-governance controls (classification tiers, role-aware redaction, tiered retention, and tamper-evident audit) for production-readiness privacy planning
```

Author: Applied in `0d4009ef` (switched to parentheses). Thanks!

> ---

Author: Fixed all flagged em dashes and the `for example` comma in `0d4009ef`. Thanks!

| Internal      | Non-personal but not for public release                           | Access scoping                                          |
| PII           | Identifies or relates to a person                                 | Minimization, access scoping, retention limits          |
| Sensitive-PII | Special-category data (health, biometric, precise location)       | Redaction, strict access, DPIA trigger, short retention |
| Regulated     | Data subject to a specific regime (HIPAA, sector retention rules) | Regime-specific obligations and evidence                |

## Obligations mapping

Map each classified element to the obligations it triggers, then to a concrete control:

- Minimization and lawful basis or consent — collect only what the purpose needs; record the basis.
- Redaction — role-aware masking of personal data in outputs and logs.
- Retention and deletion — tiered schedule with an enforced deletion mechanism.
- Breach handling — detection, notification threshold, and response path.
- Subject access and erasure — how a request is served and evidenced.

> - Audit — tamper-evident record of who did what, when, and to which data.

Reviewer: These six bullets use em dashes; a colon reads the same and matches the sibling references.

```suggestion
- Minimization and lawful basis or consent: collect only what the purpose needs; record the basis.
- Redaction: role-aware masking of personal data in outputs and logs.
- Retention and deletion: tiered schedule with an enforced deletion mechanism.
- Breach handling: detection, notification threshold, and response path.
- Subject access and erasure: how a request is served and evidenced.
- Audit: tamper-evident record of who did what, when, and to which data.
```

Author: Applied your suggestion in `0d4009ef` — colons read the same and match the siblings. Thanks!

Record each obligation's status as `Present`, `Partial`, or `Missing` with a source-control reference so the reviewer can assert readiness.

## Role-aware redaction

> Redaction is a role decision, not a blanket rule. State explicitly who is redacted and who is retained for each output surface. For example, a public or cross-tenant export may redact impacted-person PII while retaining operator identifiers needed for accountability. Never specify "redact everything" — that hides the accountability trail as well as the sensitive data.

Reviewer: Em dash in the final sentence; a semicolon keeps the meaning.

```suggestion
Redaction is a role decision, not a blanket rule. State explicitly who is redacted and who is retained for each output surface. For example, a public or cross-tenant export may redact impacted-person PII while retaining operator identifiers needed for accountability. Never specify "redact everything"; that hides the accountability trail as well as the sensitive data.
```

Author: Applied in `0d4009ef`. Thanks!

| Tier    | Contents                           | Window (example) | Mechanism                    |
|---------|------------------------------------|------------------|------------------------------|
| Full    | Complete record including PII      | 30 days          | Hard delete or crypto-shred  |
| Summary | Aggregated or de-identified record | 7 years          | Retain minimized fields only |

Anchor windows to the applicable regime or contractual requirement; mark proposed windows as assumptions to validate.

## Tamper-evident audit and evidence bundles

An audit trail that can be edited is not evidence. Require tamper-evidence and define what each entry captures and how a per-record evidence bundle is produced.

- Tamper-evidence: append-only or hash-chained log so entries cannot be silently altered.
- Entry captures: actor identity, timestamp, context, and stage or state change.
- Evidence bundle: for a regulated export (for example FOIA or litigation hold), the set of audit entries and source records for one subject, plus a verification step that proves the chain is intact.

## Suggested use

> Use this reference during Phase 4 (Controls) and Phase 5 (Impact) to specify concrete data-governance controls, and during a release-readiness review to mark each obligation `Present` / `Partial` / `Missing` against what the codebase actually implements.

Reviewer: This file has no `## Source attribution` section. All six sibling references in this folder carry one, and the licensing posture asks every `references/*.md` to cite its upstream source URLs in the file. The content here is original synthesis, so this is a consistency point rather than a copyright one. Suggested closing sections:

```suggestion
Use this reference during Phase 4 (Controls) and Phase 5 (Impact) to specify concrete data-governance controls, and during a release-readiness review to mark each obligation `Present` / `Partial` / `Missing` against what the codebase actually implements.

## Planning synthesis

These controls are a planning synthesis derived from the referenced privacy standards and are not a substitute for legal advice.

## Source attribution

Obligations anchor to the standards packages indexed in `SKILL.md`:

- NIST Privacy Framework: https://www.nist.gov/privacy-framework
- NISTIR 8062: https://doi.org/10.6028/NIST.IR.8062
- GDPR: https://gdpr-info.eu
- CCPA/CPRA: https://oag.ca.gov/privacy/ccpa
- OWASP Top 10 Privacy Risks: https://owasp.org/www-project-top-10-privacy-risks/
```

Author: Good catch on the consistency gap. Added both `## Planning synthesis` and `## Source attribution` per your suggestion in `0d4009ef`. Thanks!

| PII           | Identifies or relates to a person                                 | Minimization, access scoping, retention limits          |
| Sensitive-PII | Special-category data (health, biometric, precise location)       | Redaction, strict access, DPIA trigger, short retention |
| Regulated     | Data subject to a specific regime (HIPAA, sector retention rules) | Regime-specific obligations and evidence                |

## Obligations mapping

Map each classified element to the obligations it triggers, then to a concrete control:

- Minimization and lawful basis or consent: collect only what the purpose needs; record the basis.
- Redaction: role-aware masking of personal data in outputs and logs.
- Retention and deletion: tiered schedule with an enforced deletion mechanism.
- Breach handling: detection, notification threshold, and response path.
- Subject access and erasure: how a request is served and evidenced.
- Audit: tamper-evident record of who did what, when, and to which data.

Record each obligation's status as `Present`, `Partial`, or `Missing` with a source-control reference so the reviewer can assert readiness.

## Role-aware redaction

> Redaction is a role decision, not a blanket rule. State explicitly who is redacted and who is retained for each output surface. For example, a public or cross-tenant export may redact impacted-person PII while retaining operator identifiers needed for accountability. Never specify "redact everything"; that hides the accountability trail as well as the sensitive data.

Reviewer 2: I think there is a gap which redacts subject PII but explicitly retains operator identifies in the output. This may leak operator PII into public/cross-tenant surfaces. For example, an exported report could show `operator-ref: OP-7F31`. An authorized investigator could resolve that reference to the operator’s identity through the restricted audit system.

**Suggested fix:**

```suggestion
Redaction is a role decision, not a blanket rule. For each output surface, state which subject and operator identifiers are visible, pseudonymized, or redacted. Public and cross-tenant outputs should redact or pseudonymize both subject and operator identifiers unless the recipient has a documented need, authorization, and lawful basis. Preserve the operator's direct identity in the access-controlled audit trail and link external outputs to it through a non-identifying event or case reference. Output redaction must not delete or rewrite the underlying audit evidence.
```

Author: Good catch, thank you. You're right that retaining operator identifiers on public/cross-tenant surfaces leaks operator PII. Updated so redaction treats subject and operator identifiers symmetrically: external outputs redact or pseudonymize both unless the recipient has a documented need, authorization, and lawful basis; the operator's real identity is preserved only in the access-controlled audit trail and linked from external outputs via a non-identifying event/case reference; and redaction must not delete or rewrite the underlying audit evidence. Pushed in b65adcbd. Thanks!

## Tiered retention

Retention is tiered, not binary. Capture the full-retention window and the summary-retention window separately, with the deletion or downgrade mechanism for each tier.

| Tier    | Contents                           | Window (example) | Mechanism                    |
|---------|------------------------------------|------------------|------------------------------|
| Full    | Complete record including PII      | 30 days          | Hard delete or crypto-shred  |
| Summary | Aggregated or de-identified record | 7 years          | Retain minimized fields only |

Anchor windows to the applicable regime or contractual requirement; mark proposed windows as assumptions to validate.

## Tamper-evident audit and evidence bundles

An audit trail that can be edited is not evidence. Require tamper-evidence and define what each entry captures and how a per-record evidence bundle is produced.

> - Tamper-evidence: append-only or hash-chained log so entries cannot be silently altered.

Reviewer 2: Some thoughts.. a hash chain doesn't by itself prevent silent alteration. Anyone with write access to the store can rewrite an entry and recompute every subsequent hash, producing a fresh, internally-consistent chain. Without an independent anchor (a signature or checkpoint held outside the system) and a verification step, the rewrite is undetectable. "Cannot be silently altered" is therefore an overclaim — exactly the kind of thing that undermines an evidence bundle in a FOIA/litigation context.

**Suggested fix:**
```suggestion
- Tamper-evidence: use immutable or write-once storage, or cryptographically chain entries and periodically sign or anchor checkpoints in an independently protected system. Restrict and audit administrative access, protect signing keys, define retention enforcement, and verify the chain and anchors when producing an evidence bundle.
```

Author: Agreed, "cannot be silently altered" was an overclaim: a hash chain alone can be rewritten and recomputed by anyone with write access. Reworded to require immutable/write-once storage, or chained entries plus a periodically signed or anchored checkpoint held in an independently protected system, along with restricted/audited admin access, protected signing keys, and a chain-and-anchor verification step when producing an evidence bundle. Pushed in b65adcbd. Thanks!
